kale/pipeline/mpca_trainer.py

```diff
@@ -0,0 +1,159 @@
+        """Trainer of pipeline: MPCA->Feature selection->Classifier
+
+        Args:
+            classifier (str, optional): Classifier for training. Options: support vector machine (svc) or
+                logistic regression (lr). Defaults to 'svc'.
+            classifier_params (dict, optional): Parameters of classifier. Defaults to 'auto'.
+            mpca_params (dict, optional): Parameters of Multi-linear PCA. Defaults to None.
+            n_features (int, optional): Number of features for feature selection. Defaults to None.
+            search_params (dict, optional): Parameters of grid search. Defaults to None.
+
+        """
+        if classifier not in ['svc', 'lr']:
+            error_msg = "Valid classifier should be 'svc' or 'lr', but given %s" % classifier
+            logging.error(error_msg)
+            raise ValueError(error_msg)
+
+        self.classifier = classifier
+        # init mpca object
+        if mpca_params is None:
+            self.mpca_params = default_mpca_params
+        else:
+            self.mpca_params = mpca_params
+        self.mpca = MPCA(**self.mpca_params)
+        # init feature selection parameters
+        self.n_features = n_features
+        self.feature_order = None
+        # init classifier object
+        if search_params is None:
+            self.search_params = default_search_params
+        else:
+            self.search_params = search_params
+
+        self.auto_classifier_param = False
+        if classifier_params == "auto":
+            self.auto_classifier_param = True
+            clf_param_gird = classifiers[classifier][1]
+            self.grid_search = GridSearchCV(classifiers[classifier][0](),
+                                            param_grid=clf_param_gird,
+                                            **self.search_params)
+            self.clf = None
+        elif isinstance(classifier_params, dict):
+            self.clf = classifiers[classifier][0](**classifier_params)
+        else:
+            error_msg = "Invalid classifier parameter type"
+            logging.error(error_msg)
+            raise ValueError(error_msg)
+
+    def fit(self, x, y):
+        """Fit a pipeline with the given data x and labels y
+
+        Args:
+            x (array-like tensor): input data, shape (n_samples, I_1, I_2, ..., I_N)
+            y (array-like): data labels, shape (n_samples, )
+
+        Returns:
+            self
+        """
+        # fit mpca
+        self.mpca.fit(x)
+        self.mpca.set_params(**{"return_vector": True})
+        x_proj = self.mpca.transform(x)
+
+        # feature selection
+        if self.n_features is None:
+            self.n_features = x_proj.shape[0]
+            self.feature_order = self.mpca.idx_order
+        else:
+            f_score, p_val = f_classif(x_proj, y)
+            self.feature_order = (-1 * f_score).argsort()
+        x_proj = x_proj[:, self.feature_order][:, :self.n_features]
```

Comment:
Will a new name be better for selected features from x_proj?

Reply:
Is x_train better?

Reply:
Who can tell the difference? Isn't x also x_train?